### ISSUE TITLE
[Feature] Support iOS 14 limited access permission in permission check

### ISSUE THREAD

**Issue by kuhnroyal** [CaiJingLong/flutter_photo_manager#385]
**Is your feature request related to a problem? Please describe.**

I want to display a small banner to the user if the granted permission on iOS 14 is limited to some photos/albums.
Currently it is only possible to determine if it is granted or not.

**Describe the solution you'd like**

Something like this would allow this and is backwards compatible.
```dart
enum PhotoPermission { none, full, limited }

Future<bool> requestPermission() async {
    final permission = await requestPermissionExtended();
    return permission == PhotoPermission.full || permission = PhotoPermission.limited;
}

Future<PhotoPermission> requestPermissionExtended() async {
    final permission = await _channel.invokeMethod("requestPermission");
    return PhotoPermission.values[permission];
}
```

Of course the iOS plugin side needs to be changed to fit this.

**Comment by Artem2908** [comment]
> **Is your feature request related to a problem? Please describe.**
> 
> I want to display a small banner to the user if the granted permission on iOS 14 is limited to some photos/albums.
> Currently it is only possible to determine if it is granted or not.
> 
> **Describe the solution you'd like**
> 
> Something like this would allow this and is backwards compatible.
> 
> ```dart
> enum PhotoPermission { none, full, limited }
> 
> Future<bool> requestPermission() async {
>     final permission = await requestPermissionExtended();
>     return permission == PhotoPermission.full || permission = PhotoPermission.limited;
> }
> 
> Future<PhotoPermission> requestPermissionExtended() async {
>     final permission = await _channel.invokeMethod("requestPermission");
>     return PhotoPermission.values[permission];
> }
> ```
> 
> Of course the iOS plugin side needs to be changed to fit this.

Could you tell me in more detail how to properly use your version?

**Comment by lyio** [comment]
@Artem2908 the proposed solution would provide a additional way of retrieving permission information.

Currently, the method `requestPermission` in the plugin only tells you, whether the user granted permission to the gallery or not.

In iOS14, Apple added a third possibility. Now the user can only grant permissions for a number of hand selected photos. The proposed `requestPermissionExtended` method would be able to tell you that, so that you can display a warning or what have you in your App, reminding the user of why they do not see all their pictures in the gallery.

Usage would be as simple as 

```dart
  final permission = await  PhotoManager.requestPermissionExtended();
  switch(permission) {
    case PhotoPermission.limited:
   case PhotoPermission.full:
    // permission is granted, do whatever you like
    break;
  case PhotoPermission.none:
    // permission is denied
    break;
}
```

I hope, that answers your question.

**Comment by yanivshaked** [comment]
@CaiJingLong Any roadmap to handle this issue?

**Comment by lyio** [comment]
I'll have another look at my PR and bring it up to date shortly.
Thomas Fiedler, M.A, B.Sc.
Könneritzstr. 27
D-04229 Leipzig



Am Do., 13. Mai 2021 um 10:58 Uhr schrieb Yaniv Shaked <
***@***.***>:

**Comment by yanivshaked** [comment]
@lyio you should also add the ability to call `PHPhotoLibrary.shared().presentLimitedLibraryPicker` so that you can allow asking the user to select more images.

**Comment by lyio** [comment]
See #495 for details regarding this feature request. @CaiJingLong has already implemented this feature there.

**Comment by AlexV525** [comment]
Resolved in #495 .
